watchtower/wtclient/addr_iterator.go

```diff
@@ -162,6 +162,10 @@ func (a *addressIterator) next(lock bool) (net.Addr, error) {
 	a.mu.Lock()
 	defer a.mu.Unlock()
 
+	if a.currentTopAddr == nil {
+		return nil, ErrAddressesExhausted
+	}
+
```

Comment:
This block can go away if we move the `a.currentTopAddr = a.currentTopAddr.Next()` into the for-loop.